front/lib/actions/mcp_internal_actions/servers/schedules_management.ts

```diff
@@ -0,0 +1,502 @@
+        if (contextResult.isErr()) {
+          return contextResult;
+        }
+        const { userId, workspaceId, agentConfiguration } = contextResult.value;
+
+        const scheduleResult = await TriggerResource.fetchScheduleByIdForEditor(
+          auth,
+          scheduleId,
+          {
+            agentConfigurationId: agentConfiguration.sId,
+            editorId: userId,
+          }
+        );
+        if (scheduleResult.isErr()) {
+          return new Err(new MCPError(scheduleResult.error.message));
+        }
+
+        let scheduleUpdate:
+          | { cron: string; timezone: string; naturalLanguage: string }
+          | undefined;
+
+        if (schedule) {
+          const resolvedTimezone = getUserTimezone(agentLoopContext);
+
+          if (!resolvedTimezone) {
+            return new Err(new MCPError("Provide a timezone"));
+          }
+
+          const cronResult = await generateCronRule(auth, {
+            naturalDescription: schedule,
+            defaultTimezone: resolvedTimezone,
+          });
+
+          if (cronResult.isErr()) {
+            return new Err(
+              new MCPError(
+                `Unable to understand the schedule "${schedule}". Please try rephrasing (e.g., "every weekday at 9am", "every Monday at 10am").`
+              )
+            );
+          }
+
+          scheduleUpdate = {
+            cron: cronResult.value.cron,
+            timezone: cronResult.value.timezone,
+            naturalLanguage: schedule,
+          };
+        }
+
+        const updateResult = await TriggerResource.update(auth, scheduleId, {
+          name,
+          enabled,
+          customPrompt: prompt,
+          ...(scheduleUpdate && {
+            configuration: {
+              cron: scheduleUpdate.cron,
+              timezone: scheduleUpdate.timezone,
+            },
+            naturalLanguageDescription: scheduleUpdate.naturalLanguage,
+          }),
+        });
+
+        if (updateResult.isErr()) {
+          return new Err(
+            new MCPError(
+              `Failed to update schedule: ${updateResult.error.message}`
+            )
+          );
+        }
+
+        getStatsDClient().increment("tools.schedules_management.updated", 1, [
+          `workspace_id:${workspaceId}`,
+          `agent_id:${agentConfiguration.sId}`,
+        ]);
+
+        const changedFields = [
+          name !== undefined && "name",
+          schedule !== undefined && "schedule",
+          prompt !== undefined && "prompt",
+          enabled !== undefined && (enabled ? "enabled" : "disabled"),
+        ]
+          .filter(Boolean)
+          .join(", ");
+
+        return new Ok([
+          {
+            type: "text" as const,
+            text: `Updated schedule (changed: ${changedFields}).\n\n${renderSchedule(updateResult.value)}`,
+          },
+        ]);
+      }
+    )
+  );
+
+  server.tool(
+    "delete_schedule",
+    "Permanently delete a schedule.",
```

Comment:
don't really see the need of this method either tbh, if we make the parallel with agent management, we can just create agents, and that's all

i don't see a world where one would just ask the agent to delete its own schedule, but i might be too 2024-coded, i sometimes forget we live in the future